### PR TITLE
huggers now can NOT be thrown over barricades

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request

This PR makes huggers unable to be thrown over barricades

## Why It's Good For The Game

Hugger spam is nearly impossible to stop because shooting 5 huggers in a second isnt practical

## Changelog
:cl:
balance: huggers can now no longer be thrown over barricades, and will drop and go idle at your feet.
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
BEGONE, HUGGERS HIDING BEHIND CADES FACING SOUTH!!!!

Now carriers can't contribute to FOB camping.

**Comment by Naaanii** [comment]
CLM!
Carrier Lives Matter!

**Comment by joookschad** [comment]
CLM
Just shoot at the huggers, its really not that hard.
"Hugger spam is nearly impossible to stop because shooting 5 huggers in a second isnt practical"
It's literally not hard to kill 5 huggers when you have a few marines at the FOB.

**Comment by Naaanii** [comment]
Huggers can also be meleed when thrown the jump can be dodged, resetting the jump timer.

I do agree that hiding them against South facing cades, is pretty cheese though.

**Comment by zerosexualpartners** [comment]
Practically no one uses that feature, so, I don't see the reason why that feature must be deleted.
There is more serious problems like tackle spam capturing, overpowered boiler with tweaked cooldowns, or hivemind wall spawn on FOB zone.

**Comment by Hughgent** [comment]
In my opinion, huggers should be able to be thrown over `unwired` barricades.

Please punish marines for being unprepared. Don't just stop huggers always.

File: code/modules/mob/living/carbon/xenomorph/facehuggers.dm [hunk 1]
```diff
@@ -232,6 +233,17 @@
...
 
+/obj/item/clothing/mask/facehugger/proc/check_barricade(datum/source, atom/target, range, thrower, spin)
+	var/mob/living/carbon/xenomorph/X = usr
+	var/list/turf_between = getline(X,target)
+	for(var/turf/T in turf_between)
+		var/obj/structure/barricade/found = locate() in T
+		if(found)
+			if(!found.density)
+				continue
+			to_chat(usr,"<span class='danger'>The facehugger isnt able to be thrown over the barricade!</span>")
```
Comment (Hughgent) [inline code/modules/mob/living/carbon/xenomorph/facehuggers.dm:244]: Please make this more 'lore firendly'

Perhaps "you're underhand throw slams the hugger into the [found.name]"

Comment (MacBlaze1) [reply code/modules/mob/living/carbon/xenomorph/facehuggers.dm:244]: Hugh it just drops at your feet, and I would have to edit throw_ats base atom proc to change that

**Comment by MacBlaze1** [comment]
the build failed due to some dumb unreachable code that isnt broken

**Comment by SplinterGP** [comment]
> Practically no one uses that feature, so, I don't see the reason why that feature must be deleted, its like a cool trick for experienced players.
> There is more serious problems like tackle spam capturing, overpowered boiler with tweaked cooldowns, or hivemind wall spawn on FOB zone.

Everyone does use that when theyre close to FOB, AND saying just shoot hugger is easy when it aint you and huggers are bugged as hell amd never behave consistently, also as if theyre not just going to throw more after you kill them

**Comment by TiviPlus** [comment]
No.
Either fix the "Inconsistent huggers" or literalyl walk up and click on them it isnt hard

**Pull request closed without merging**